ext/pcntl/pcntl.c

```diff
@@ -125,6 +125,16 @@ static zend_class_entry *QosClass_ce;
 
 #define LONG_CONST(c) (zend_long) c
 
+#if defined(__linux__)
```

Comment:
good intention but not quite. see musl define all [here](https://git.musl-libc.org/cgit/musl/tree/include/sys/wait.h#n17).

what you can do however is to add the header presence check in ext/pcntl/config.m4

AC_CHECK_HEADER([linux/wait.h])
which should create the following preprocessor

HAVE_LINUX_WAIT_H

(so before recompiling you need to start fresh including doing ./buildconf)

After that, you should be able to test P_PIDFD.

Reply:
Good point and good idea, made the change.